docs/release/release-process.md

```diff
@@ -0,0 +1,293 @@
+# Release Process
+
+This document is meant to be a complete end-to-end guide for how to release new versions of software
+for Crossplane and its related projects.
+
+## tl;dr Process Overview
+
+All the details are available in the sections below, but we'll start this guide with a very high
+level sequential overview for how to run the release process.
+
+1. **feature freeze**: Merge all completed features into master branches of all repos to begin
+   "feature freeze" period
+1. **RC tag**: Run tag pipeline to tag the release candidate (RC) build in master for each repo
+1. **branch**: Create a new release branch using the GitHub UI for each repo
+1. **release stacks**: Run the release process for each **stack** that we maintain
+    1. **test** Test builds from the release branch, fix any critical bugs that are found
+    1. **version**: Update all version information in the stack metadata and docs in the release
```

Comment:
Do we want to mention the integration test change that has to happen here as well? I can look into fixing that for future releases.